### PR TITLE
feat(survey): homepage survey 2024

### PR DESCRIPTION
**Pull request by fiji-flo**
## Summary

We want user input for our homepage. So we put a survey up.

---

### Before

image

### After

image

---

## How did you test this change?
Locally and testing

### CODE REVIEW

File: client/src/document/on-github.tsx [hunk 1]
```diff
@@ -4,13 +4,13 @@ export function OnGitHubLink({ doc }: { doc: Doc }) {
   return (
     <div id="on-github" className="on-github">
       <SourceOnGitHubLink doc={doc}>
-        {doc.locale !== "de"
+        {doc.locale !== "fr"
```
Comment (caugner) [inline client/src/document/on-github.tsx:7]: These changes need to be reverted.

File: client/src/ui/molecules/document-survey/surveys.ts [hunk 2]
```diff
@@ -100,4 +89,14 @@ export const SURVEYS: Survey[] = [
     start: 0,
     end: Infinity,
   },
+  {
+    key: SurveyKey.HOMEPAGE_FEEDBACK_2024,
+    bucket: SurveyBucket.HOMEPAGE_FEEDBACK_2024,
+    show: (doc: { mdn_url: string }) => /[^/]+\/$/i.test(doc.mdn_url),
+    src: "https://survey.alchemer.com/s3/8075407/MDN-Homepage-Improvements",
+    teaser: "We are refreshing out homepage and would love",
+    question: "your input",
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/surveys.ts:98]: Typo. Is this the final copy?

```suggestion
    teaser: "We are refreshing our homepage and would love",
    question: "your input",
```

File: client/src/ui/molecules/document-survey/surveys.ts [hunk 3]
```diff
@@ -1,18 +1,17 @@
-import { Doc } from "../../../../../libs/types/document";
 import { survey_duration, survey_rates } from "../../../env";
 
 export interface Survey {
   key: SurveyKey;
   bucket: SurveyBucket;
-  show: (doc: Doc) => boolean;
+  show: (doc: { mdn_url: string }) => boolean;
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/surveys.ts:6]: Semantically better:

```suggestion
  show: (doc: Partial<Doc, "mdn_url">) => boolean;
```

**Review by caugner — CHANGES_REQUESTED** [review]

File: client/src/document/on-github.tsx (hunk 1 above)
Comment (fiji-flo) [reply client/src/document/on-github.tsx:7]: Ah, I was looking at my TS issue there and added it 🙈

File: client/src/ui/molecules/document-survey/index.tsx [hunk 4]
```diff
@@ -59,7 +58,7 @@ function SurveyDisplay({
   survey,
   force,
 }: {
-  doc: Doc;
+  doc: { mdn_url: string };
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/index.tsx:61]:
```suggestion
  doc: Pick<Doc, "mdn_url">;
```

File: client/src/ui/molecules/document-survey/index.tsx [hunk 5]
```diff
@@ -12,7 +11,7 @@ import { SURVEY } from "../../../telemetry/constants";
 
 const FORCE_SURVEY_PREFIX = "#FORCE_SURVEY=";
 
-export function DocumentSurvey({ doc }: { doc: Doc }) {
+export function DocumentSurvey({ doc }: { doc: { mdn_url: string } }) {
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/index.tsx:14]:
```suggestion
export function DocumentSurvey({ doc }: { doc: Pick<Doc, "mdn_url"> }) {
```

File: client/src/ui/molecules/document-survey/index.tsx [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import React from "react";
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/index.tsx:1]:
```suggestion
import React from "react";
import { Doc } from "../../../../../libs/types/document";
```

File: client/src/ui/molecules/document-survey/surveys.ts [hunk 7]
```diff
@@ -100,4 +90,14 @@ export const SURVEYS: Survey[] = [
     start: 0,
     end: Infinity,
   },
+  {
+    key: SurveyKey.HOMEPAGE_FEEDBACK_2024,
+    bucket: SurveyBucket.HOMEPAGE_FEEDBACK_2024,
+    show: (doc: Pick<Doc, "mdn_url">) => /[^/]+\/$/i.test(doc.mdn_url),
+    src: "https://survey.alchemer.com/s3/8075407/MDN-Homepage-Improvements",
+    teaser: "We are refreshing our homepage and would love",
+    question: "your input",
```
Comment (caugner) [inline client/src/ui/molecules/document-survey/surveys.ts:99]: Nit: Usually we put the first question of the survey in the `question` field.

**Review by caugner — APPROVED** [review]
Didn't test, overall LGTM, just nits.

File: client/src/ui/molecules/document-survey/surveys.ts (hunk 7 above)
Comment (caugner) [reply client/src/ui/molecules/document-survey/surveys.ts:99]: @fiji-flo Is this ok for screen readers?

image

**Pull request merged**